src/libraries/System.Private.CoreLib/src/System/SearchValues/ProbabilisticWithAsciiCharSearchValues.cs

```diff
@@ -134,7 +138,9 @@ internal override int IndexOfAnyExcept(ReadOnlySpan<char> span)
 
         internal override int LastIndexOfAny(ReadOnlySpan<char> span)
         {
-            if (IndexOfAnyAsciiSearcher.IsVectorizationSupported && span.Length >= Vector128<short>.Count)
+            // We check whether the last character is ASCII before calling into IndexOfAnyAsciiSearcher
+            // in order to minimize the overhead this fast-path has on non-ASCII texts.
+            if (IndexOfAnyAsciiSearcher.IsVectorizationSupported && span.Length >= Vector128<short>.Count && char.IsAscii(span[^1]))
```

Comment:
Does the bounds check get eliminated here?

Reply:
should be with https://github.com/dotnet/runtime/pull/84213, if not - please let me know 🙂

Reply:
It is
![image](https://github.com/dotnet/runtime/assets/25307628/fd1abe95-ed61-42ae-beb3-a4662b249e83)